cmd/fleet/handleAck.go

```diff
@@ -130,11 +128,7 @@ func (ack *AckT) processRequest(zlog zerolog.Logger, w http.ResponseWriter, r *h
 
 	zlog = zlog.With().Int("nEvents", len(req.Events)).Logger()
 
-	if err = ack.handleAckEvents(r.Context(), zlog, agent, req.Events); err != nil {
-		return err
-	}
-
-	resp := AckResponse{"acks"}
+	resp := ack.handleAckEvents(r.Context(), zlog, agent, req.Events)
```

Comment:
This change seems to be breaking to me. Being that an error will no longer be returned from `handleAck` doesn't that mean the status code for the whole response will always be a 200, where previously it could have been a 400 or something.

Say an older not updated Elastic Agent goes to Ack a policy and that fails, this change will have it return a 200 and the Elastic Agent will not try to do the action again.

I think this flow needs to be changed to compute the worst HTTP status code from the responses and return that value as the response code for the Ack.

Reply:
I like #1. That is because a newer Elastic Agent will know to read the body and do more with it, where the older one will not need to it will just keep working how it always has.

Even in the case that a newer Elastic Agent talks to an older Fleet Server (which we don't support), it could be smart and ignore an empty response body.